crates/extensions/c8y_mapper_ext/src/operations/handler.rs

```diff
@@ -254,8 +258,15 @@ struct RunningOperation {
     status: String,
 }
 
-fn is_clearing_message(message: &MqttMessage) -> bool {
-    message.payload().is_empty() && message.retain && message.qos == QoS::AtLeastOnce
+// TODO: logic of which status transitions are valid should be defined in tedge_api and be
+// considered together with custom statuses of custom workflows
+fn is_operation_status_transition_valid(previous: &str, next: &str) -> bool {
+    #[allow(clippy::match_like_matches_macro)]
+    match (previous, next) {
+        ("successful", "executing") => false,
+        ("failed", "executing") => false,
```

Comment:
[successful -> failed] and [failed -> successful] cases should be also forbidden. To be precise, any state transitions from the final state (successful/failed) is forbidden. So, something like below is more accurate. (not sure if this is valid Rust, however, hope you understand what I wanted to say)

```rust
        ("successful", _) => false,
        ("failed", _) => false,
```

Reply:
Fixed in 0ebe371d15